### ISSUE TITLE
Problem: extra structMap breaks dip upload

### ISSUE THREAD

**Issue by jhsimpson** [artefactual/archivematica#822]
Archivematica 1.7.0 has a new feature that allows unique identifiers to be assigned to all files and folders in a SIP, even when some of the directories have no contents.

Empty directories are described in a new, 2nd logical structmap (labelled 'normative directory structure') that is added to the METS file.  

The feature is described here:

https://github.com/artefactual-labs/archivematica-acceptance-tests/blob/master/features/core/pid-binding.feature#L17-L29

This new logical structmap breaks dip upload in Atom 2.4 and earlier.  It should be possible to revert to the previous (1.6.x) behaviour and not include the new structmap if it is not required.  Possibly setting a new optional parameter passed to the create METS microservice?

**Comment by jrwdunham** [comment]
I think the solution would be to add a new boolean option like `--create-normative-structmap` to MCPClient/lib/clientScripts/archivematicaCreateMETS2.py with a default value (of `False`?). 

Then we would want this to be user-configurable in some way. The easy way would be to have normative structmap creation turned off by default and then have admin users use MySQL to modify the `arguments` column of the `StandardTasksConfigs` table with execute = `createMETS_v2.0`. 

Better would be to allow runtime configuration via the processing config, but that would likely require a migration and modifications to the workflow...

**Comment by sromkey** [comment]
As we were just discussing in Slack, a user-configurable micro-service named something like "Document empty directories?" would be ideal.

**Comment by sromkey** [comment]
This has been closed by #833 .

**Comment by currmie** [comment]
Kicked off transfers in augustus.qa, am17x.qa, and centos17x.qa.
Successfully sent DIPs from augustus to atom-24rc3-test server, but Rsync quits unexpectedly for transfers that started in am17x.qa and centos17x.qa.

Here's the error message:
Standard error (stderr)

upload-qubit.py: ERROR     2018-02-10 03:05:47,165  archivematica.upload.qubit:log:64:  [uploadDIP] Target: file-1
upload-qubit.py: ERROR     2018-02-10 03:05:47,165  archivematica.upload.qubit:log:64:  [uploadDIP] rsync --protect-args -rltz -P --chmod=ugo=rwX /var/archivematica/sharedDirectory/watchedDirectories/uploadDIP/loris3-2145cef3-9f6b-48a4-9ae9-f884aed6833d 158.69.72.17:/tmp
upload-qubit.py: ERROR     2018-02-10 03:05:47,166  archivematica.upload.qubit:log:64:  [uploadDIP] Rsync output is being saved in /tmp/tmpxireOP
[uploadDIP] Rsync quit unexpectedly (exit 12), the upload script will be stopped here

**Comment by sromkey** [comment]
@currmie , just confirming- the fix is to make sure your METS does NOT have the extra structMap by choosing "no" to Document empty directories- is this what you did?

**Comment by currmie** [comment]
Hi @sromkey - yes, I did what you outlined as one part of my two-part test:

For the first part, I ran a standard transfer to verify that the METS did NOT include the second structMap labeled "Normative Directory Structure". That part passed.

For the second part of my test, I ran another transfer with the same objects but chose YES to document empty directories. I wanted to verify that at AIP re-ingest the new logical structMap would be parsed to re-create the empty directories per the development documentation > Metadata elements > METS. At AIP reingest is where I began running into problems. I guess the breakage is likely unrelated to the fix for issue #822 and is a symptom of a different problem. (In my latest attempt at AIP reingest, the transfer fails at the Microservice: Assign file UUIDs and checksums. Prior to that, problems came up at DIP upload. Not sure why.)

For simplicity's sake, I can mark this issue as 'verified' if the fix is solely to address the first part of my test.
